### PR TITLE
Refactor MB06

### PR DESCRIPTION
**Pull request by malmans2**
@oceandie if you are done with MB06, I think this can be merged into #33.
Needs a good check, but I think it's consistent with the implementation with loops.
I used [this notebook](https://gist.github.com/malmans2/906d6e8699722849706175f5e7bdc4d0) to test.

### CODE REVIEW

File: pydomcfg/utils.py
Comment (oceandie) [inline pydomcfg/utils.py:164]: I personally would put an error: from personal experience, usually it is sufficient tiincrease the tolerance of a couple of orders, an many tines it is relating with the precision of your data ... in this way we force the user to be aware and maybe make the input more consistent precision wise

File: pydomcfg/utils.py
Comment (oceandie) [inline pydomcfg/utils.py:143]: Very minor point but just for the logical flow: I would call mb06 zr instead, since what we compute here is strictly the slope paramater ... then, in the line below, we refine the computation only in ocean areas ...

File: pydomcfg/utils.py
Comment (oceandie) [inline pydomcfg/utils.py:92]: I would add something like: This is the xarray version of NEMO implementation ....

**Review by oceandie — APPROVED** [review]

**Comment by oceandie** [comment]
Very clear implementation, thanks @malmans2 !

File: pydomcfg/utils.py
Comment (malmans2) [reply pydomcfg/utils.py]: OK, changed to error! Maybe we should decrease the default number of iterations... I'd get pretty mad if I wait hours to get an error and no return :) 
Anyways, not urgent.

File: pydomcfg/utils.py
Comment (malmans2) [reply pydomcfg/utils.py]: Reworded, although I didn't explicitly mention xarray as we use it pretty much everywhere and it should be clear from the docs. Also I removed the units, any units should work fine.

**Comment by malmans2** [comment]
> Very clear implementation, thanks @malmans2 !

Thanks! BTW, eventually I think we should make this function public. But no need to do it right now, we can wait until we're ready to merge `dev_sco`.

File: pydomcfg/utils.py
Comment (oceandie) [reply pydomcfg/utils.py]: Yes, I agree!

File: pydomcfg/utils.py
Comment (oceandie) [reply pydomcfg/utils.py]: Yes, I agree with your point ... it was just to make clear to the users (an us :) ) that this is only a more efficient version of the NEMO MB06 implementation

**Review by oceandie — APPROVED** [review]

**Comment by malmans2** [comment]
All set! Not sure if you're ready to integrate this in your `sco_dev` PR, so I'll let you merge this.

**Review by oceandie — APPROVED** [review]

**Pull request merged**
